### DISCUSSION TITLE
Your thoughts wanted: How should BurntToast work?

### DISCUSSION THREAD

**Discussion by Windos** [Windos/BurntToast#160] [Ideas]
There is a poll up on Twitter on this topic: https://twitter.com/WindosNZ/status/1372801243001720841

There's a good chance the next release of BurntToast will have a _significant_ number of breaking changes. This means it's a good time to think through how it's used.

The options I'm weighing up are:

1. Continuing as is, making heavy use of functions
2. Creating less functions and relying more on the native .NET methods
3. Creating a DSL

![Mock up of options](https://user-images.githubusercontent.com/6955786/111741838-060b5f00-88ec-11eb-9c40-3dcf957ef8ae.png)

Please let me know your thoughts on the options listed here or any alternatives.

**Comment by sgexcit** [comment]
I'm still very much learning the ropes with PowerShell, so I'm curious as to what the benefits DSL or .NET over Functions. 

I'm partial to functions at the moment because its what I know, but if either of the other two methods could spell better longevity, security, or functionality for the module I'm all for it!

**Comment by Windos** [comment]
Hello, and thank you for your contribution to BurntToast.

Today marks the official release of [BurntToast v1.0.0](https://github.com/Windos/BurntToast/releases/tag/v1.0.0). As part of this major milestone and to ensure the project starts its next chapter with a clean slate, I am declaring "issue bankruptcy."

This decision is explained in more detail in the official announcement post: [Ten Years Toasting: BurntToast Hits the Big v1!](https://toastit.dev/2025/07/22/burnttoast-v1/)

As a result, I am closing all outstanding issues and pull requests (and discussions), including this one. Please do not interpret this as a lack of interest in the problem, suggestion, or topic you've raised. This is a necessary step to clear a very old backlog and focus development efforts on the current state of the module.

**If this topic is still relevant to you:**

1.  Please **open a new discussion**. You are welcome to link back to this one for historical context.

I hope this reset will allow me to be more responsive to community feedback going forward.

Thank you for your understanding and for being part of the BurntToast journey.
